spring-web/src/main/java/org/springframework/http/converter/StringHttpMessageConverter.java

```diff
@@ -139,18 +139,18 @@ protected void writeInternal(String str, HttpOutputMessage outputMessage) throws
 	}
 
 	private Charset getContentTypeCharset(@Nullable MediaType contentType) {
-		if (contentType != null && contentType.getCharset() != null) {
-			return contentType.getCharset();
-		}
-		else if (contentType != null && contentType.isCompatibleWith(MediaType.APPLICATION_JSON)) {
-			// Matching to AbstractJackson2HttpMessageConverter#DEFAULT_CHARSET
-			return StandardCharsets.UTF_8;
-		}
-		else {
-			Charset charset = getDefaultCharset();
-			Assert.state(charset != null, "No default charset");
-			return charset;
+		if (contentType != null) {
+			Charset charset = contentType.getCharset();
+			if (charset != null) {
+				return charset;
+			} else if (contentType.isCompatibleWith(MediaType.APPLICATION_JSON)) {
```

Comment:
Please adhere to project coding standards.

```suggestion
			}
			else if (contentType.isCompatibleWith(MediaType.APPLICATION_JSON)) {
```